pkg/metadata/metadata.go

```diff
@@ -82,3 +88,43 @@ func ApplyAnnotations(c client.Client, ctx context.Context, node core.Node, anno
 	}
 	return nil
 }
+
+func ApplyDesiredVersionAnnotation(ctx context.Context, c client.Client, node core.Node, value string) error {
+	return ApplyLabelsAndAnnotations(ctx, c, node, nil, map[string]string{DesiredVersionAnnotation: value})
+}
+
+// RemoveVersionAnnotation clears the version annotation from the node object, indicating the node is not configured
+func RemoveVersionAnnotation(ctx context.Context, c client.Client, node core.Node) error {
+	if _, present := node.GetAnnotations()[VersionAnnotation]; present {
+		patchData, err := GenerateRemovePatch([]string{}, []string{VersionAnnotation})
+		if err != nil {
+			return errors.Wrapf(err, "error creating version annotation remove request")
+		}
+		err = c.Patch(ctx, &node, client.RawPatch(kubeTypes.JSONPatchType, patchData))
+		if err != nil {
+			return errors.Wrapf(err, "error removing version annotation from node %s", node.GetName())
+		}
+	}
+	return nil
+}
+
+// WaitForVersionAnnotation checks if the node object has the version annotation with the given value.
+// Waits for retry.Interval seconds and returns an error if it does not appear in that time frame.
+func WaitForVersionAnnotation(ctx context.Context, c client.Client, node core.Node, expectedVal string) error {
```

Comment:
Any reason this can't just wait until desiredVersion annotation == version annotation?

Reply:
Also, why is node being passed in when `Get` is being called to retrieve it? The node name should be the argument here.

Reply:
No strong reason for either. I've taken both suggestions, thanks